lib/cocoapods/target/pod_target.rb

```diff
@@ -200,56 +200,59 @@ def supported_test_types
     # @return [Array<Hash{Symbol => [String]}>] The vendored and non vendored framework paths
     #         this target depends upon.
     #
-    def framework_paths
-      @framework_paths ||= begin
```

Comment:
removed the `@framework_paths ||= begin` this is why the diff looks weird.

Reply:
Easier to review with https://github.com/CocoaPods/CocoaPods/pull/7000/files?w=1

Reply:
What's the purpose of removing the caching? This is kinda expensive

Reply:
So now we are parameterized whether we return paths including the ones from tests or not.

I can actually cache it where the key includes the boolean if you'd like. WDYT?

Reply:
Cache is now added back and it depends on the parameter.